### DISCUSSION TITLE
Tindie companion shield won't connect for ESPHome config upload

### DISCUSSION THREAD

**Discussion by PoebelPogge** [gcormier/megadesk#150] [Q&A] [answered]
Hey 👋 

I bought couple of days ago the companion shield from Tindie. It made the long wide way from Canada to Germany. Now I'm trying to flash the ESPHome config on it, but it constantly says that it there is no connection.
By using the CLI it says: 
`ERROR Running command failed: Failed to connect to ESP8266: No serial data received.`
Using ESPHome web it connects to the first step but then it says that:

> Failed to initialize. Try resetting your device or holding the BOOT button while selecting your serial port until it starts preparing the installation.

No matter if I hit `Prepare for first use` or if I try to upload the compiled `.bin` 

Any ideas on what I'm doing wrong?

**Comment by gcormier** [comment]
I'm not sure if ESPHome web works great with the C3 yet. There've been a few patches in the last few weeks ( eg https://github.com/esphome/esphome/releases/tag/2023.12.6 ) specifically for the C3 but I haven't had great luck. Maybe using command line will improve the results?

There are steps here
https://www.youtube.com/watch?v=hUrNMm47_AY

If you can provide more details on the steps you are taking, I can try it as well.

**Comment by PoebelPogge** [comment]
okay, following that video, I got a step further, now it says:

`ERROR Running command failed: This chip is ESP32-C3 not ESP8266`

I used the [megadesk.yml](https://github.com/gcormier/megadesk/blob/master/esphome/megadesk.yaml) from the repository.

Changing `platform: ESP8266` to `platform: ESP32-C3` leads to the following error by running: `esphome compile megadesk.yml`:

```
INFO ESPHome 2023.12.9
INFO Reading configuration megadesk.yml...
Failed config

esphome: [source <unicode string>:5]
  name: megadesk
  comment: ESPHome Device controlling Megadesk controller
  
  Unknown value 'esp32-c3', did you mean 'esp32'?.
  platform: ESP32-C3
  board: d1_mini
  includes: 
    - megadesk.h
```

**Reply by gcormier** [reply, marked as answer]
Please use https://github.com/gcormier/megadesk/blob/master/esphome/megadesk-companion.yaml

**Reply by PoebelPogge** [reply]
Awesome! Thank you so much! :)
That did the job :)

**Reply by Ragetti13** [reply]
I am having a similar issue with trying to get the first flash working. I have it  connected up to my windows PC and have downloaded all of the python and esphome repositories on powershell and both version are up to date. I've used the above link to download the referenced files from github but when trying to compile i receive this error:

```
INFO ESPHome 2024.6.6
INFO Reading configuration megadesk-companion.yaml...
Failed config

ota.unknown: [source megadesk-companion.yaml:50]

  'ota' requires a 'platform' key but it was not specified.
  password: ''
```

Any suggestions?

**Reply by gcormier** [reply]
Esphome updated some configuration requirements recently. I'll try to make an update but unsure if I will have time this week. You can use an older esphome version for now, or maybe someone else will submit a PR to update the yaml with the new requirements.

**Reply by gcormier** [reply]
Config updated for new esphome - https://github.com/gcormier/megadesk/blob/master/esphome/megadesk-companion.yaml

**Reply by Ragetti13** [reply]
Amazing, thanks Greg - That has worked and i just need to calibrate it now :-)
